pkg/yurtmanager/controller/platformadmin/platformadmin_controller.go

```diff
@@ -440,11 +441,21 @@ func (r *ReconcilePlatformAdmin) reconcileComponent(ctx context.Context, platfor
 			// Refresh the YurtAppSet according to the user-defined configuration
 			yas.Spec.WorkloadTemplate.DeploymentTemplate.Spec = *desiredComponent.Deployment
 
-			if _, ok := yas.Status.PoolReplicas[platformAdmin.Spec.PoolName]; ok {
-				if yas.Status.ReadyReplicas == yas.Status.Replicas {
-					readyDeployment = true
-					if readyDeployment && readyService {
-						readyComponent++
+			for _, summary := range yas.Status.WorkloadSummaries {
+				poolName := ""
+				for i, str := range strings.Split(summary.WorkloadName, "-") {
+					if i == len(strings.Split(summary.WorkloadName, "-"))-1 {
+						break
+					}
+					poolName += str
+					poolName += "-"
+				}
```

Comment:
It seems there are some problems to resolve nodepool name from summary.WorkloadName. please have a check.

Reply:
fix